gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/TableInfo.java

```diff
@@ -0,0 +1,741 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.client.util.Data;
+import com.google.api.services.bigquery.model.Streamingbuffer;
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.ViewDefinition;
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Lists;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Google BigQuery Table information. A BigQuery table is a standard, two-dimensional table with
+ * individual records organized in rows, and a data type assigned to each column (also called a
+ * field). Individual fields within a record may contain nested and repeated children fields. Every
+ * table is described by a schema that describes field names, types, and other information.
+ *
+ * @see <a href="https://cloud.google.com/bigquery/docs/tables">Managing Tables</a>
+ */
+public class TableInfo implements Serializable {
+
+  static final Function<Table, TableInfo> FROM_PB_FUNCTION =
+      new Function<Table, TableInfo>() {
+        @Override
+        public TableInfo apply(Table pb) {
+          return TableInfo.fromPb(pb);
+        }
+      };
+  static final Function<TableInfo, Table> TO_PB_FUNCTION =
+      new Function<TableInfo, Table>() {
+        @Override
+        public Table apply(TableInfo tableInfo) {
+          return tableInfo.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -7679032506430816205L;
+
+  /**
+   * The table type.
+   */
+  public enum Type {
```

Comment:
I feel that using a standard Java inheritance would be a nicer user experience than having a flat type with all possible setter/getter (merged from all types TABLE, EXTERNAL and VIEW).